### PR TITLE
Renames 'rotation' partial operations to 'quarterturn's.

### PR DESCRIPTION
**Pull request by MichaelQ94**
Allows specification of which face to rotate. Improves documentation.

### CODE REVIEW

File: src/lib/core/partialcubeop.ts [hunk 1]
```diff
@@ -1,45 +1,70 @@
...
+import { CubeIndex, Face } from "lib/core/cubeindex";
 
 /**
- * Computes the new (row, column) coordinates of the input CubeIndex after a
- * clockwise rotation. Does not account for movements of indices on adjacent faces.
+ * CubeIndex literals will be represented as {FRC} where:
+ * F is the face coordinate
+ * R is the row coordinate
+ * C is the column coordinate
+ */
```
Comment (MichaelQ94) [inline src/lib/core/partialcubeop.ts]: Going to move this and other docs about the cube representation to the main README at some point.

File: src/lib/core/partialcubeop.ts [hunk 2]
```diff
@@ -1,45 +1,70 @@
...
- * {1, 0}, {1, 1}, {1, 2}  ->  {2, 1}, {1, 1}, {0, 1}
- * {2, 0}, {2, 1}, {2, 2}      {2, 2}, {1, 2}, {0, 2}
+ *    {F00} {F01} {F02}      {F20} {F10} {F00}
+ *    {F10} {F11} {F12}  ->  {F21} {F11} {F01}
+ *    {F20} {F21} {F22}      {F22} {F12} {F02}
  *
- * @param cubeIndex The CubeIndex to be rotated
- * @param dimension The dimension of the cube being acted on
+ * Facelets occupying all other positions will be unaffected, including those on faces
+ * adjacent to `faceToTurn`.
```
Comment (MichaelQ94) [inline src/lib/core/partialcubeop.ts]: Going to add some tests for this.

File: src/lib/core/partialcubeop.ts [hunk 3]
```diff
@@ -1,45 +1,70 @@
...
+ * @param {number} position.row The row coordinatte of the facelet's current position
+ * @param {number} position.column The column coordinate of the facelet's current position
+ * @param {Face} faceToTurn The cube face to be rotated by this operation
+ * @param {number} dimension The dimension of the cube being acted on
  */
-export function rotateClockwise(
+export function clockwiseQuarterTurn(
   { face, row, column }: CubeIndex,
+  faceToTurn: Face,
   dimension: number,
```
Comment (MichaelQ94) [inline src/lib/core/partialcubeop.ts]: At some point this is going to be curried, but that will come after all partial operations are implemented.

File: src/lib/core/partialcubeop.ts (hunk 1 above)
Comment (itstang) [reply src/lib/core/partialcubeop.ts]: link out to a task 🙏

File: src/lib/core/partialcubeop.ts [hunk 4]
```diff
@@ -1,45 +1,70 @@
...
-    row: dimension - 1 - column,
-    column: row,
-  };
+  return face === faceToTurn
+    ? {
+        face,
+        row: dimension - 1 - column,
+        column: row,
+      }
+    : { face, row, column };
```
Comment (KeijiBranshi) [inline src/lib/core/partialcubeop.ts]: Why would the caller invoke this function with a different `faceToTurn` if it already knows the `face` on the `CubeIndex` it provided?

Will this particular line ever return anything different in the future? If not, I question the need for the extra parameter.

File: src/lib/core/partialcubeop.ts [hunk 5]
```diff
@@ -1,45 +1,70 @@
...
  *
- * @param cubeIndex The CubeIndex to be rotated
- * @param dimension The dimension of the cube being acted on
+ * Facelets occupying all other positions will be unaffected, including those on faces
+ * adjacent to `faceToTurn`.
+ *
+ * @param {CubeIndex} position The starting position of the facelet to be moved
+ * @param {Face} position.face The face coordinate of the facelet's current position
+ * @param {number} position.row The row coordinatte of the facelet's current position
+ * @param {number} position.column The column coordinate of the facelet's current position
```
Comment (KeijiBranshi) [inline src/lib/core/partialcubeop.ts]: nit: seems like this should be implied by the `@param {CubeIndex}...` above. Maybe these comments should go above the definition of `CubeIndex` itself?

File: src/lib/core/partialcubeop.ts (hunk 4 above)
Comment (MichaelQ94) [reply src/lib/core/partialcubeop.ts]: These are really utility functions to implement actual CubeOps with. The idea is to put the branching logic here in order prevent callers from having to write it themselves:

```if (cubeIndex.face === faceIWantToRotate) {
/* do partial operation */
} else {
/* do nothing */
} ```

They can just trust that partial operations themselves will only affect specific CubeIndexes and will be no-op for all others.

File: src/lib/core/partialcubeop.ts (hunk 5 above)
Comment (MichaelQ94) [reply src/lib/core/partialcubeop.ts]: Makes sense. Moved.

File: src/lib/core/partialcubeop.ts (hunk 1 above)
Comment (MichaelQ94) [reply src/lib/core/partialcubeop.ts]: Done.

File: src/lib/core/partialcubeop.test.ts [hunk 6]
```diff
@@ -49,241 +74,313 @@ const CUBE_FACE_3X3 = {
   },
 };
 
-describe("3x3 clockwise rotation", () => {
+function expectAllOtherFacesUnaffected(
+  cubeFace: typeof CUBE_FACE_3X3 | typeof CUBE_FACE_2X2,
+  operation: (cubeIndex: CubeIndex, face: Face, dimension: number) => CubeIndex,
+): void {
+  Object.entries(cubeFace)
+    .filter(([key, _]) => key !== "dimension")
```
Comment (itstang) [inline src/lib/core/partialcubeop.test.ts]: would it make sense to make the test object something like:

```js
{
  dimension: 3,
  facelets: {
    topLeft: {...}
    ...
  }
}
```

so you could just skip this filtering step?

e.g.
```js
Object.entries(cubeFace.facelets).forEach(([key, value]) => {...})
```

(I don't know if facelet is the proper nomenclature here)

also side note, technically in JS, `_` is still a real variable, you should just be able to omit it entirely if you're not using it

Comment (MichaelQ94) [reply src/lib/core/partialcubeop.test.ts]: It would make a lot of sense. Done.

File: src/lib/core/partialcubeop.test.ts [hunk 7]
```diff
@@ -3,145 +3,145 @@ import * as partial from "./partialcubeop";
...
+  });
 }
 
 describe("2x2 clockwise quarter turn", () => {
   it("sends top left corner to top right corner", () => {
     expect(
       partial.clockwiseQuarterTurn(
-        CUBE_FACE_2X2.topLeft,
-        CUBE_FACE_2X2.topLeft.face,
+        CUBE_FACE_2X2.positions.topLeft,
```
Comment (itstang) [inline src/lib/core/partialcubeop.test.ts]: not necessary, but you can pull out positions in the beginning of the describe block so you don't have to key in so deep every time.

```ts
describe("2x2 clockwise quarter turn", () => {
	const positions = CUBE_FACE_2X2.positions;

	it("sends top left corner to top right corner", () => { ... }
}
```

Comment (MichaelQ94) [reply src/lib/core/partialcubeop.test.ts]: Also done.

**Pull request merged**